backend/capellacollab/projects/crud.py

```diff
@@ -42,6 +43,11 @@ def create_project(
     db: Session, name: str, description: str | None = None
 ) -> DatabaseProject:
     slug = slugify(name)
+    if (
+        db.query(DatabaseProject).filter(DatabaseProject.slug == slug).first()
+        is not None
+    ):
+        raise HTTPException(409, "Slug already used.")
```

Comment:
This should not happen in the `crud` module. The `crud` module is used to access the database. 

When I'm calling the function from another context, I wouldn't expect a `HTTPException`. I would either expect a custom (and documentation) error. 

The alternative would be to have another module with validators (e.g. `validators.py`) to validate the non-existence of a project, which can be executed before we actually create the model in the database.

Reply:
I though that since the error was clearly SQL-related (violation of unique constraint), I left it as is, and handle it in the route.

Reply:
I meant that I was going to throw the SQL exception, the past tense referred to the correction I just had done ^^ Anyways, it’s now as you describe in your answer.

Reply:
Looks good!